Assets/BossRoom/Scripts/Client/UI/UnityServicesUIHandler.cs

```diff
@@ -27,21 +27,24 @@ void ServiceErrorHandler(UnityServiceErrorMessage error)
             var errorMessage = error.Message;
             if (error.AffectedService == UnityServiceErrorMessage.Service.Lobby)
             {
-                if ((error.OriginalException as LobbyServiceException).Reason == LobbyExceptionReason.LobbyConflict)
+                switch ((error.OriginalException as LobbyServiceException).Reason)
                 {
-                    // LobbyConflict can have multiple causes. Let's add other solutions here if there's other situations that arise for this.
-                    errorMessage += "\nSee logs for possible causes and solution.";
-                    Debug.LogError($"Got service error {error.Message} with LobbyConflict. Possible conflict cause: Trying to play with two builds on the " +
-                                   $"same machine. Please use command line arg '{ClientMainMenuState.AuthProfileCommandLineArg} someName' to set a different auth profile.\n");
+                    case LobbyExceptionReason.LobbyConflict:
+                        // LobbyConflict can have multiple causes. Let's add other solutions here if there's other situations that arise for this.
+                        Debug.LogError($"Got service error {error.Message} with LobbyConflict. Possible conflict cause: Trying to play with two builds on the " +
+                            $"same machine. Please use command line arg '{ClientMainMenuState.AuthProfileCommandLineArg} someName' to set a different auth profile.\n");
+                        PopupPanel.ShowPopupPanel("Service error", error.Message + "\nSee logs for possible causes and solution.");
+                        break;
+                    case LobbyExceptionReason.LobbyFull:
+                        PopupPanel.ShowPopupPanel("Failed to join lobby", "Lobby is full and can't accept more players");
+                        break;
                 }
             }
-
-            PopupPanel.ShowPopupPanel("Service error", errorMessage);
```

Comment:
Wait, wouldn't this hide other error messages? This was here to at least show "something" when we had an unhandled error

Reply:
Fixed this, we're no longer hiding other messages.